### PR TITLE
Proposal:  support edge nodes report events

### PR DESCRIPTION
**Pull request by llmmqq8023**
**What type of PR is this?**

/kind design

**What this PR does / why we need it:**

Proposal: support edge nodes report events

issue: https://github.com/kubeedge/kubeedge/issues/5595

### CODE REVIEW

**Review by Shelley-BaoYue — COMMENTED** [review]
Looks good to me : ) 
Please add a head at the top of proposal file, format like: 
```
---
title: xxx
authors:
- "@xxx"
  approvers: 
  creation-date: 2024-07-12
  last-updated: 2024-07-12
  status: implementable
---
```

**Review by Shelley-BaoYue — COMMENTED** [review]
Commit name is used to let others to known what work has done in this commit. You can just squash these two commit to one, or name the second commit with `add title for event report proposal`
image

File: docs/proposals/support-edge-nodes-report-events.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+The project plan is very similar to the `patchPod` procedure elaborated in Preliminaries. Below is the (expected) information flow chart of edge events, where PLEG is taken as an example. 
+
+![flow chart of edge events](../images/proposals/support-edge-nodes-report-events.png)
+
+Items marked red are the information flow of events. Items marked orange are components to be edited. The specific procedures are:
+1. Pod Lifecycle Event Generator (PLEG), which is deployed in Kubelet, generates an event, and then the EventBridge API implemented in metaclient is called to report this event to the cloud. 
+2. Metaclient wraps this event into beehive message and puts it into the specific beehive channel. 
+3. Metamanager grabs the event message and give it to processEvent handler. 
+4. The event is stored in the edge (optional).
```
[the remark this answers is not in the document]
Comment (fisherxu) [reply docs/proposals/support-edge-nodes-report-events.md:256]: Now it's not stored in db :)

**Review by fisherxu — COMMENTED** [review]
The config field `SendEvent` or `ReportEvent` should be added to the proposal :)

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**